src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    internal static class GeneratedCodeUtilities
+    {
+        private static readonly string[] s_autoGeneratedStrings = new[] { "<autogenerated", "<auto-generated" };
+
+        internal static bool IsGeneratedSymbolWithGeneratedCodeAttribute(
```

Comment:
We can have the API return more fine-grained information.  Can either of you enumerate which cases are which?  I'll have the API return an enum to specify what kind of generated file it is, and we can have our features then appropriately decide on what to do.